models/siegert_neuron.h

```diff
@@ -0,0 +1,327 @@
+
+Description:
+
+ siegert_neuron is an implementation of a rate model with the
+ non-linearity given by the gain function of the
+ leaky-integrate-and-fire neuron with delta or exponentially decaying
+ synapses. The model can be used for a mean-field analysis of spiking
+ networks.
+
+ The model supports connections to other rate models with zero
+ delay, and uses the secondary_event concept introduced with the
+ gap-junction framework.
+
+Parameters:
+
+ The following parameters can be set in the status dictionary.
+
+ rate                double - Rate (1/s)
+ tau                 double - Time constant in ms.
+ mean                double - Additional constant input
+
+ The following parameters can be set in the status directory and are
+ used in the evaluation of the gain function. Parameters as in
+ iaf_psc_exp/delta.
+
+ tau_m               double - Membrane time constant in ms.
+ tau_syn             double - Time constant of postsynaptic currents in ms.
+ t_ref               double - Duration of refractory period in ms.
+ theta               double - Threshold relative to resting potential in mV.
+ V_reset             double - Reset relative to resting membrane potential in
+                              mV.
+
+Notes:
+
+References:
+
+ [1] Hahne, J., Dahmen, D., Schuecker, J., Frommer, A.,
+ Bolten, M., Helias, M. and Diesmann, M. (2017).
+ Integration of Continuous-Time Dynamics in a
+ Spiking Neural Network Simulator.
+ Front. Neuroinform. 11:34. doi: 10.3389/fninf.2017.00034
+
+ [2] Schuecker, J., Diesmann, M. and Helias, M. (2015).
+ Modulated escape from a metastable state driven by colored noise.
+ Physical Review E 92:052119
+
+ [3] Hahne, J., Helias, M., Kunkel, S., Igarashi, J.,
+ Bolten, M., Frommer, A. and Diesmann, M. (2015).
+ A unified framework for spiking and gap-junction interactions
+ in distributed neuronal network simulations.
+ Front. Neuroinform. 9:22. doi: 10.3389/fninf.2015.00022
+
+Sends: DiffusionEvent
+
+Receives: DiffusionEvent, DataLoggingRequest
+
+Author: Jannis Schuecker, David Dahmen, Jan Hahne
+SeeAlso: diffusion_connection
+*/
+
+class siegert_neuron : public Archiving_Node
+{
+
+public:
+  typedef Node base;
+
+  siegert_neuron();
+  siegert_neuron( const siegert_neuron& );
+
+  /**
+   * Import sets of overloaded virtual functions.
+   * @see Technical Issues / Virtual Functions: Overriding, Overloading, and
+   * Hiding
+   */
+  using Node::handle;
+  using Node::sends_secondary_event;
+
+  void handle( DiffusionEvent& );
+  void handle( DataLoggingRequest& );
+
+  port handles_test_event( DiffusionEvent&, rport );
+  port handles_test_event( DataLoggingRequest&, rport );
+
+  void
+  sends_secondary_event( DiffusionEvent& )
+  {
+  }
+
+  void get_status( DictionaryDatum& ) const;
+  void set_status( const DictionaryDatum& );
+
+private:
+  void init_state_( const Node& proto );
+  void init_buffers_();
+  void calibrate();
+  bool update_( Time const&, const long, const long, const bool );
```

Comment:
@heplesser Of course you are right about this. However my impression was that this is quite easy to understand (the occurrences of update_ in lines 260 and 272 are in fact the only ones) and that the "threshold" for comments in the NEST code in general is rather high.

I added a documentation to `update_`.